src/app/dashboard/import-export/import-modal/import-modal.component.ts

```diff
@@ -42,21 +43,41 @@ export class ImportModalComponent implements OnInit {
   setImportFile($event) {
     if ($event.target.files) {
       if ($event.target.files.length !== 0) {
-        let regex = /.json$/;
+        let regex = /.gz$/;
         if (regex.test($event.target.files[0].name)) {
-          this.fileReference = $event;
-          this.validFile = true;
-        } else {
-          let fr: FileReader = new FileReader();
-          fr.readAsText($event.target.files[0]);
-          fr.onloadend = (e) => {
+          let blob = new Blob([$event.target.files[0]], { type: 'application/octet-stream' });
+          let arrayBuffer;
+          let result;
+          let fileReader = new FileReader();
+          fileReader.onload = () => {
```

Comment:
I'd advise taking a second look at this block of fileReader calls from ln 52 on, debugging, making sure that the file content is coming in the correct format/encoding as it should be. Give it a look, reach out if you don't find the issue.